server/src/ClientToDBAPI/UsersRoute/mainRoute.ts

```diff
@@ -1,32 +1,62 @@
 
+import User from '../..//Models/User/User';
 import { Router, Request, Response } from 'express';
-import { GET_IS_USER_ACTIVE } from "../../DBService/Users/Query";
+import { GET_IS_USER_ACTIVE, GET_USER_BY_ID, GET_ALL_GROUP_USERS } from '../../DBService/Users/Query';
 
 const usersRoute = Router();
 
 usersRoute.get('/userActivityStatus', (request: Request, response: Response) => {
-    graphqlRequest(GET_IS_USER_ACTIVE, response.locals, {id: response.locals.user.id})
-    .then((result:any) => {
-        if (result.data?.userById)
-            response.send(result.data?.userById);
-        else
-            response.status(400).send(`Couldn't find the user nor get its status`);
-    })
-    .catch(error => response.status(500).send('Error while trying to fetch isActive user status'))
+    graphqlRequest(GET_IS_USER_ACTIVE, response.locals, { id: response.locals.user.id })
+        .then((result: any) => {
+            if (result.data?.userById)
+                response.send(result.data?.userById);
+            else
+                response.status(400).send(`Couldn't find the user nor get its status`);
+        })
+        .catch(error => response.status(500).send('Error while trying to fetch isActive user status'))
 })
 
 usersRoute.post('/updateIsUserActive', (request: Request, response: Response) => {
-    graphqlRequest(UPDATE_IS_USER_ACTIVE, response.locals, {id: response.locals.user.id, isActive: request.body.isActive})
-       .then((result:any) => {
-        if(result.data.updateUserById)
-            response.send(result.data.updateUserById.user);
-        else
-            response.status(400).send(`Couldn't find the user nor update the status`);
-
-       })
-       .catch(error => response.status(500).send('Error while trying to activate / deactivate user'))
+    graphqlRequest(UPDATE_IS_USER_ACTIVE, response.locals, { id: response.locals.user.id, isActive: request.body.isActive })
+        .then((result: any) => {
+            if (result.data.updateUserById)
+                response.send(result.data.updateUserById.user);
+            else
+                response.status(400).send(`Couldn't find the user nor update the status`);
+
+        })
+        .catch(error => response.status(500).send('Error while trying to activate / deactivate user'))
 })
 
+usersRoute.get('/getUserById', (request: Request, response: Response) => {
+    graphqlRequest(GET_USER_BY_ID, response.locals, { id: request.query.id })
+        .then((result: any) => response.send(result.data));
+});
+
+usersRoute.post('/changeInvestigator', (request: Request, response: Response) => {
+    graphqlRequest(UPDATE_INVESTIGATOR, response.locals, { id: request.query.id })
+        .then((result: any) => response.send(result.data));
+});
+
+usersRoute.get('/getAllGroupUsers', (request: Request, response: Response) => {
+    graphqlRequest(GET_ALL_GROUP_USERS, response.locals, { investigationGroup: +request.query.investigationGroup })
```

Comment:
add check in server for all admin-only routes that user is admin.
consider adding a middleware that fetches the admin status according to the user id
(in response.locals) and add it to those routes

Reply:
dont need to. Request will only be sent to server when it's the admin

Reply:
the route is still exposed